src/app.ts

```diff
@@ -12,15 +12,74 @@ interface TemperatureSummary {
   low: number
   average: number
 }
+/* dictionary of dictionarys for the summarys, i use a counter for  */
+interface IndexedSumary {
+  [city: string]: {
+    [time: string]: {
+      summary: TemperatureSummary
+      count: number
+    }
+  }
+}
+const temperatures: IndexedSumary = {}
 
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  //add here your code
+  /*we iterate over all the readings, if there isn't a valid entry for a key, it creates it
+  each read updates the values if it is necesary
+  
+  i used for of instead of reduce because its faster as far as I have been able to research
+  and 1 loop does less comparations than 3*/
+  for (const read of readings) {
+    const readTimeToString: string = read.time.toString()
+    if (temperatures[read.city] == undefined) {
+      temperatures[read.city] = {}
+    }
+    if (temperatures[read.city][readTimeToString] == undefined) {
+      temperatures[read.city][readTimeToString] = {
+        summary: {
+          first: read.temperature, // if the key doesn't had an assigned value then it's the first measure
+          last: read.temperature,
+          high: read.temperature,
+          low: read.temperature,
+          average: read.temperature,
+        },
+        count: 1,
+      }
+    } else {
+      const tempAux: TemperatureSummary =
```

Comment:
You shouldn't need to add this type here. If your typescript is correct it should be inferred

Reply:
I got it, I put it preventively, thanks for the suggestion

Reply:
I like to let typescript do as much work for me as possible